web-react/src/pages/directory/display/DetailsFellowship.jsx

```diff
@@ -34,24 +34,38 @@ const DetailsFellowship = () => {
     breadcrumb = [fellowship]
   }
 
-  const lastFilledServices = history?.services.map((service) => service.week)
-  const lastFilledBanking = history?.services.map(
-    (service) => service.bankingProof
+  const lastFilled = history?.services.map(
+    ({ bankingProof, noServiceReason, week }) => ({
+      bankingProof,
+      noServiceReason,
+      week,
+    })
   )
 
-  const check = last3Weeks()?.map((week, i) => {
-    if (lastFilledServices?.includes(week)) {
-      return {
-        number: week,
-        filled: true,
-        banked:
-          lastFilledBanking?.length && (lastFilledBanking[i] ? true : false),
+  const check = last3Weeks()?.map((number) => {
+    if (lastFilled?.some((thing) => thing.week === number)) {
```

Comment:
Please name your variables better. What is thing? Inside? 

It's not easy to understand

Reply:
Changed